client/components/copy-button/test/index.test.tsx

```diff
@@ -0,0 +1,75 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+/**
+ * Internal dependencies
+ */
+import { CopyButton } from '..';
+
+describe( 'CopyButton', () => {
+	it( 'renders the button correctly', () => {
+		const { container: copyButtonContainer } = render(
+			<CopyButton
+				textToCopy="test_bank_reference_id"
+				label="Copy bank reference ID to clipboard"
+			/>
+		);
+
+		expect( copyButtonContainer ).toMatchSnapshot();
+	} );
+
+	describe( 'when the button is clicked', () => {
+		beforeEach( () => {
+			jest.useFakeTimers();
+		} );
+
+		afterEach( () => {
+			jest.useRealTimers();
+		} );
+
+		it( 'copies the text to the clipboard and shows copied state', async () => {
+			render(
+				<CopyButton
+					textToCopy="test_bank_reference_id"
+					label="Copy bank reference ID to clipboard"
+				/>
+			);
+
+			const button = screen.queryByRole( 'button', {
+				name: /Copy bank reference ID to clipboard/i,
+			} );
+
+			if ( ! button ) {
+				throw new Error( 'Button not found' );
+			}
+
+			//Mock the clipboard API
+			Object.assign( navigator, {
+				clipboard: {
+					writeText: jest.fn().mockResolvedValueOnce( undefined ),
+				},
+			} );
+
+			await act( async () => {
+				fireEvent.click( button );
+			} );
+
+			expect( navigator.clipboard.writeText ).toHaveBeenCalledWith(
+				'test_bank_reference_id'
+			);
+			expect( button ).toHaveClass( 'state--copied' );
+
+			act( () => {
+				jest.advanceTimersByTime( 2000 );
```

Comment:
Ah, so it seems that the jest JSDOM env won't be running CSS animations, but this can be manually triggered using `fireEvent`:

```suggestion
				fireEvent.animationEnd( button );
```

Then we also won't need to override the jest timer setup and teardown.

Reply:
Thanks Eric!